docker_context/Dockerfile

```
@@ -0,0 +1,52 @@
+FROM ubuntu:22.04
+
+ENV LANG C.UTF-8
+
+# Work around apt-get hanging during install because it tries to query the user
+ARG DEBIAN_FRONTEND=noninteractive
+
+# Upgrade system and install base packages
+RUN apt-get update --fix-missing\
+  && apt-get -qy upgrade \
+  && apt-get -qy install --no-install-recommends \
+    build-essential \
+    cmake  \
+    clang-format \
+    clang-tidy \
+    cppcheck \
+    graphviz \
+    # Ninja
+    ninja-build \
+    libncurses5 \
+    libc++abi-dev \
+    libc++-dev \
+    # plant uml
+	  plantuml \
+    # python pip
+	  python3-pip \

```

```suggestion
    plantuml \
    # python pip
    python3-pip \
```

The editorconfig-checker should complain if tabs are used instead of spaces. I like the idea of adding an `.editorconfig` file, but we should also add a CI job for checking it.

---

good idea - i added the editorconfig check an updated the Dockerfile accordingly